velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -240,6 +240,9 @@ SubstraitVeloxExprConverter::toVeloxExpr(
           BaseVector::wrapInConstant(1, 0, literalsToArrayVector(substraitLit));
       return std::make_shared<const core::ConstantTypedExpr>(constantVector);
     }
+    case ::substrait::Expression_Literal::LiteralTypeCase::kDate:
+      return std::make_shared<core::ConstantTypedExpr>(
+          variant(substraitLit.date()));
```

Comment:
This needs to be `variant(Date(substraitLit.date()))`

Reply:
Ah this is the missing piece then.